### ISSUE TITLE
MSI Creator 17 B11UH-405BE i9-11900H

### ISSUE THREAD

**Issue by EnesPalit** [YoCodingMonster/OpenFreezeCenter#178]
TL;DR: Not working on _Kali Linux 2023.3_

First of all:

Thank you for this awesome project. My SSD got bad sectors and I couldn't boot into Windows anymore. I got my whole drive recovered on Ubuntu with the use of gddrescue. But when I first tried it, my laptop was overheating and and automatically shutting down. So I needed to get the fan working in order to get it cooled while doing extensively data recovery. So I found your project. Now, for some reason, on _Ubuntu_ I can't mount my partitions where the _Windows_ data is on. I tried _Kali Linux_ and on there I was able to mount my _Windows_ partition. But on there, I'm not getting your project to work. 

I successfully got it working on my **MSI Creator 17 B11UH-405BE i9-11900H** running _Ubuntu 23.10_.

However, when I run it on _Kali Linux 2023.3_ it doesn't work. I did the same steps as I did on _Ubuntu 23.10_. But it doesn't generate the _config.py_ file.

It is the same laptop but it doesn't have the same outcome on both different Operating Systems.

I tried to copy the _config.py_ file from my Ubuntu drive to my Kali Linux drive but I still have an error. Should I copy the ic debug folder too?

**Comment by YoCodingMonster** [comment]
Hi @EnesPalit
Thanks for Tring this application. I am glad this is helping you!

For Kali, you need to run command ```grep ec_sys /lib/modules/$(uname -r)/modules.builtin```
If the output is ```kernel/drivers/acpi/ec_sys.ko```
Then you need to do following things.
- Add ```ec_sys.write_support=1``` in ```/etc/default/grub```
- Update grub: ```update-grub```(Manjaro) or ```grub-mkconfig -o /boot/grub/grub.cfg```(Arch)
- Reboot.

Try this and see if it works in kali linux!

**Comment by EnesPalit** [comment]
Thank you! I will definitely try this, 12 hours from now.

**Comment by YoCodingMonster** [comment]
If That doesn't work, then you need to do.

- Copy both ```ec_sys.conf``` files provided with same path in the attached zip.
- ```etc/modprobe.d/ec_sys.conf```
- ```etc/modules-load.d/ec_sys.conf```
- Reboot OR ```modprobe ec_sys write_support=1```
[etc.zip](https://github.com/YoCodingMonster/OpenFreezeCenter/files/13402075/etc.zip)

**Comment by EnesPalit** [comment]
```
modprobe ec_sys write_support=1
modprobe: FATAL: Module ec_sys not found in directory /lib/modules/6.3.0-kali1-amd64
```

**Comment by EnesPalit** [comment]
```
sudo update-grub
/usr/sbin/grub-mkconfig: 42: /etc/default/grub: ec_sys.write_support=1: not found
```

**Comment by EnesPalit** [comment]
```
sudo ls -a /sys/module/acpi/parameters/
.
..
acpica_version
aml_debug_output
ec_busy_polling
ec_delay
ec_event_clearing
ec_freeze_events
ec_max_queries
ec_no_wakeup
ec_polling_guard
ec_storm_threshold
immediate_undock
sleep_no_lps0

```

**Comment by EnesPalit** [comment]
`ls -a ~/Documents/OFC
.
..
OFC.py
bin
include
lib
lib64
nohup.out
pyvenv.cfg
`

**Comment by EnesPalit** [comment]
```
Traceback (most recent call last):
  File "/home/newuser/Documents/OFC/OFC.py", line 59, in <module>
    CONFIG_FILE = open(PATH_TO_CONFIG, "r")
                  ^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/home/newuser/Documents/OFC/config.py'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/newuser/Documents/OFC/OFC.py", line 64, in <module>
    LINE_NO = "PROFILE = 1\nAUTO_SPEED = [["+str(ECT.read(0x72, 1))+", "+str(ECT.read(0x73, 1))+", "+str(ECT.read(0x74, 1))+", "+str(ECT.read(0x75, 1))+", "+str(ECT.read(0x76, 1))+", "+str(ECT.read(0x77, 1))+", "+str(ECT.read(0x78, 1))+"], ["+str(ECT.read(0x8a, 1))+", "+str(ECT.read(0x8b, 1))+", "+str(ECT.read(0x8c, 1))+", "+str(ECT.read(0x8d, 1))+", "+str(ECT.read(0x8e, 1))+", "+str(ECT.read(0x8f, 1))+", "+str(ECT.read(0x90, 1))+"]]"
                                                 ^^^^^^^^^^^^^^^^^
  File "/home/newuser/Documents/OFC/lib/python3.11/site-packages/ECTweaker.py", line 39, in read
    with open(EC_IO_FILE,'r+b') as file:
         ^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/sys/kernel/debug/ec/ec0/io'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/newuser/Documents/OFC/OFC.py", line 76, in <module>
    import config
ModuleNotFoundError: No module named 'config'
Traceback (most recent call last):
  File "/home/newuser/Documents/OFC/OFC.py", line 59, in <module>
    CONFIG_FILE = open(PATH_TO_CONFIG, "r")
                  ^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/home/newuser/Documents/OFC/config.py'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/newuser/Documents/OFC/OFC.py", line 64, in <module>
    LINE_NO = "PROFILE = 1\nAUTO_SPEED = [["+str(ECT.read(0x72, 1))+", "+str(ECT.read(0x73, 1))+", "+str(ECT.read(0x74, 1))+", "+str(ECT.read(0x75, 1))+", "+str(ECT.read(0x76, 1))+", "+str(ECT.read(0x77, 1))+", "+str(ECT.read(0x78, 1))+"], ["+str(ECT.read(0x8a, 1))+", "+str(ECT.read(0x8b, 1))+", "+str(ECT.read(0x8c, 1))+", "+str(ECT.read(0x8d, 1))+", "+str(ECT.read(0x8e, 1))+", "+str(ECT.read(0x8f, 1))+", "+str(ECT.read(0x90, 1))+"]]"
                                                 ^^^^^^^^^^^^^^^^^
  File "/home/newuser/Documents/OFC/lib/python3.11/site-packages/ECTweaker.py", line 39, in read
    with open(EC_IO_FILE,'r+b') as file:
         ^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/sys/kernel/debug/ec/ec0/io'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/newuser/Documents/OFC/OFC.py", line 76, in <module>
    import config
ModuleNotFoundError: No module named 'config'

```

**Comment by EnesPalit** [comment]
```
sudo ls -a /sys/kernel/debug
.
..
acpi
asoc
bdi
block
bluetooth
cec
clear_warn_once
clk
cxl
devfreq
device_component
devices_deferred
dma_buf
dma_pools
dmaengine
dri
dynamic_debug
energy_model
extfrag
fault_around_bytes
frontswap
gpio
hid
ieee80211
intel_lpss
intel_powerclamp
kprobes
kvm
lru_gen
lru_gen_full
mce
mei0
mmc0
opp
pinctrl
pkg_temp_thermal
pm_genpd
pmc_core
pwm
ras
regmap
regulator
resctrl
sched
slab
sleep_time
sof
soundwire
split_huge_pages
sunrpc
suspend_stats
swiotlb
thunderbolt
tracing
ttm
usb
wakeup_sources
x86
zswap
```

**Comment by YoCodingMonster** [comment]
> If That doesn't work, then you need to do.
> 
> 
> 
> - Copy both ```ec_sys.conf``` files provided with same path in the attached zip.
> 
> - ```etc/modprobe.d/ec_sys.conf```
> 
> - ```etc/modules-load.d/ec_sys.conf```
> 
> - Reboot OR ```modprobe ec_sys write_support=1```
> 
> [etc.zip](https://github.com/YoCodingMonster/OpenFreezeCenter/files/13402075/etc.zip)
> 
> 

Did this worked?

**Comment by EnesPalit** [comment]
> > If That doesn't work, then you need to do.
> > 
> > 
> > 
> > - Copy both ```ec_sys.conf``` files provided with same path in the attached zip.
> > 
> > - ```etc/modprobe.d/ec_sys.conf```
> > 
> > - ```etc/modules-load.d/ec_sys.conf```
> > 
> > - Reboot OR ```modprobe ec_sys write_support=1```
> > 
> > [etc.zip](https://github.com/YoCodingMonster/OpenFreezeCenter/files/13402075/etc.zip)
> > 
> > 
> 
> Did this worked?

No it did not work.

**Comment by EnesPalit** [comment]
> ```
> modprobe ec_sys write_support=1
> modprobe: FATAL: Module ec_sys not found in directory /lib/modules/6.3.0-kali1-amd64
> ```

See this too

**Comment by YoCodingMonster** [comment]
> > ```
> > modprobe ec_sys write_support=1
> > modprobe: FATAL: Module ec_sys not found in directory /lib/modules/6.3.0-kali1-amd64
> > ```
> 
> See this too

-  need to install kali and test this myself.
will update you in a couple of days

**Comment by EnesPalit** [comment]
> > > ```
> > > modprobe ec_sys write_support=1
> > > modprobe: FATAL: Module ec_sys not found in directory /lib/modules/6.3.0-kali1-amd64
> > > ```
> > 
> > See this too
> 
> -  need to install kali and test this myself.
> will update you in a couple of days

Okay thank you!

I used the Kali Linux  [https://www.kali.org/get-kali/#kali-live ](Live Boot) version. I downloaded the [https://cdimage.kali.org/kali-2023.3/kali-linux-2023.3-live-amd64.iso](Kali Linux 2023.3 Live amd64 Point release live image) and installed it on my laptop from that live image.

**Comment by arnavoke** [comment]
i ran into the same issue, actually a lot, needed to copy some import dependencies into local include folders, but after making config part working i encountered this exact same issue, with same errors.
have you found any solution on this?

**Comment by EnesPalit** [comment]
> > > ```
> > > modprobe ec_sys write_support=1
> > > modprobe: FATAL: Module ec_sys not found in directory /lib/modules/6.3.0-kali1-amd64
> > > ```
> > 
> > See this too
> 
> -  need to install kali and test this myself.
> will update you in a couple of days

Any updates? Will it be possible to get it working on Kali 2023.4?
